Packs/DeCYFIR/Integrations/decyfir/decyfir.py

```
@@ -0,0 +1,483 @@
+                {
+                    "type": "incident_source_from",
+                    "value": source_brand
+                }
+            ]
+        }
+
+        return return_data
+
+    def prepare_incidents_for_attack_surface(self, json_data, alert_type: str, alert_subtype: str) -> List:
+        try:
+            incidents_json = []
+            for json_ in json.loads(json_data):
+                severity = self.get_severity(json_.get("risk_score"))
+                ip = json_.get("ip")
+                details = str(json.dumps(json_))
+                date_val = json_.get("alert_created_date")
+                uid = json_.get("uid")
+
+                domain: str = ""
+                if json_.get("sub_domain") is not None and json_.get("sub_domain") != "":
+                    domain = json_.get("sub_domain")
+                if json_.get("top_domain") is not None and json_.get("top_domain") != "":
+                    if domain.__eq__("") or domain == "":
+                        domain = json_.get("top_domain")
+                    else:
+                        domain = domain + ", " + json_.get("top_domain")
+
+                name = ""
+                if domain is not None and domain != "":
+                    name = "DOMAIN : {}".format(domain)
+
+                if ip is not None and ip != "":
+                    if name is not None and name != "":
+                        name = name + "\n IP: {}".format(ip)
+                    else:
+                        name = "IP: {}".format(ip)
+
+                incident_json = self.prepare_incident_json(LABEL_DECYFIR, alert_type, alert_subtype,
+                                                           name, date_val, severity, details, uid)
+                incidents_json.append(incident_json)
+
+            return incidents_json
+
+        except Exception as e:
+            print("Exception when calling prepare_incidents_for_attack_surface=> : %s\n" % e)
+            return_error(str(e))
```

Use raise instead of print and return_error (with a clear message), it will be caught in the main function.